pkg/controller/hvpa/hvpa_controller.go

```diff
@@ -951,47 +1089,139 @@ func getWeightedRequests(vpaStatus *vpa_api.VerticalPodAutoscalerStatus, hvpa *a
-					(overrideScaleUpStabilization || lastScaleTimeDuration > scaleUpStabilizationWindow) {
-					// If the difference is greater than minimum delta, AND
-					// scale stabilization window is expired
-					log.V(2).Info("VPA", "Scaling up", "CPU", "Container", container.Name)
-					newPodSpec.Containers[id].Resources.Requests[corev1.ResourceCPU] = weightedCPU.DeepCopy()
-					resourceChange = true
-				} else if scaleDownEnaled && diffCPU.Sign() < 0 && negDiffCPU.Cmp(*scaleDownMinDeltaCPU) > 0 &&
-					lastScaleTimeDuration > scaleDownStabilizationWindow {
-					log.V(2).Info("VPA", "Scaling down", "CPU", "Container", container.Name)
-					newPodSpec.Containers[id].Resources.Requests[corev1.ResourceCPU] = weightedCPU.DeepCopy()
-					resourceChange = true
+				if vpaWeight == 0 {
+					outTargetWeight[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+					blockedByWeight = true
+
+				} else if diffCPU.Sign() > 0 {
+					if hpaScaleOutLimited == false || *hvpa.Spec.Vpa.UpdatePolicy.UpdateMode == autoscalingv1alpha1.UpdateModeOff {
+						outTargetUpdatePolicy[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+						blockedByUpdatePolicy = true
+
+					} else if overrideScaleUpStabilization == false && lastScaleTimeDuration < scaleUpStabilizationWindow {
+						outTargetStabilizationWindow[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+						blockedByStabilizationWindow = true
+
+					} else if diffCPU.Cmp(*scaleUpMinDeltaCPU) < 0 {
+						outTargetMinChanged[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+						blockedByMinChange = true
+
+					} else {
+						log.V(2).Info("VPA", "Scaling up", "CPU", "Container", container.Name)
+						newPodSpec.Containers[id].Resources.Requests[corev1.ResourceCPU] = weightedCPU.DeepCopy()
+						// Override VPA status in outVpaStatus with weighted value
+						outTarget[corev1.ResourceCPU] = weightedCPU.DeepCopy()
+						resourceChange = true
+					}
+				} else if diffCPU.Sign() < 0 {
+					if *hvpa.Spec.Vpa.UpdatePolicy.UpdateMode == autoscalingv1alpha1.UpdateModeOff ||
+						*hvpa.Spec.Vpa.UpdatePolicy.UpdateMode == autoscalingv1alpha1.VpaUpdateModeScaleUp {
+						outTargetUpdatePolicy[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+						blockedByUpdatePolicy = true
+
+					} else if lastScaleTimeDuration < scaleDownStabilizationWindow {
+						outTargetStabilizationWindow[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+						blockedByStabilizationWindow = true
+
+					} else if negDiffCPU.Cmp(*scaleDownMinDeltaCPU) < 0 {
+						outTargetMinChanged[corev1.ResourceCPU] = rec.Target.Cpu().DeepCopy()
+						blockedByMinChange = true
+
+					} else {
+						log.V(2).Info("VPA", "Scaling down", "CPU", "Container", container.Name)
+						newPodSpec.Containers[id].Resources.Requests[corev1.ResourceCPU] = weightedCPU.DeepCopy()
+						// Override VPA status in outVpaStatus with weighted value
+						outTarget[corev1.ResourceCPU] = weightedCPU.DeepCopy()
+						resourceChange = true
```

Comment:
We should ideally still push to `outTargetWeight` `vpaCPUTarget` if it is not equal to `weightedCPU`. But we can pick this in a later issue.

Reply:
Resolving in favour of #20.